bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/AndroidDebugBridgeDiscoveryService.java

```diff
@@ -0,0 +1,171 @@
+                    if (!discoveryRunning)
+                        break;
+                    if (!(inetAddress instanceof Inet4Address))
+                        continue;
+                    if (inetAddress.getHostAddress().equals(LOCAL_INTERFACE_IP))
+                        continue;
+                    String[] ipParts = inetAddress.getHostAddress().split("\\.");
+                    for (var i = this.discoveryIpRangeMin; i <= this.discoveryIpRangeMax; i++) {
+                        if (!discoveryRunning)
+                            break;
+                        ipParts[3] = Integer.toString(i);
+                        var currentIp = String.join(".", ipParts);
+                        try {
+                            var currentAddress = InetAddress.getByName(currentIp);
+                            logger.debug("address: {}", currentIp);
+                            if (currentAddress.isReachable(reachableTimeoutMs)) {
+                                logger.debug("Reachable ip: {}", currentIp);
+                                var retries = 0;
+                                while (retries < MAX_RETRIES) {
+                                    try {
+                                        discoverWithADB(currentIp);
+                                    } catch (IOException e) {
+                                        var message = e.getMessage();
+                                        if (message != null && message.contains("rejected by remote peer")) {
+                                            retries++;
+                                            logger.debug("retrying - pending {}", MAX_RETRIES - retries);
+                                            Thread.sleep(5000);
+                                            continue;
+                                        }
+                                        throw e;
+                                    } catch (AndroidDebugBridgeDevice.AndroidDebugBridgeDeviceReadException e) {
+                                        retries++;
+                                        if (retries < MAX_RETRIES) {
+                                            logger.debug("retrying - pending {}", MAX_RETRIES - retries);
+                                            continue;
+                                        }
+                                        throw e;
+                                    }
+                                    break;
+                                }
+                            }
+                        } catch (IOException | AndroidDebugBridgeDevice.AndroidDebugBridgeDeviceException
+                                | AndroidDebugBridgeDevice.AndroidDebugBridgeDeviceCryptographyException
+                                | AndroidDebugBridgeDevice.AndroidDebugBridgeDeviceNotConnectedException
+                                | AndroidDebugBridgeDevice.AndroidDebugBridgeDeviceReadException e) {
```

Comment:
If you don't need to distinguish between the exception types, you could use the same type with a different message.